develop/parachains/testing/benchmarking.md

```diff
@@ -120,52 +132,74 @@ Before running the benchmarking tool, you must integrate benchmarks with your ru
 
-1. Run `build` with the feature flag included
+1. Run `build` with the feature flag included:
 
     ```bash
     cargo build --features runtime-benchmarks --release
     ```
 
-2. Once compiled, run the benchmarking tool to measure extrinsic weights
+2. Create a `weights.rs` file in your pallet's `src/` directory. This file will store the auto-generated weight calculations:
+```bash
+touch weights.rs
+```
 
-    ```sh
-    ./target/release/INSERT_NODE_BINARY_NAME benchmark pallet \
+3. Before running the benchmarking tool, you'll need a template file that defines how weight information should be formatted. Download the official template from the Polkadot SDK repository and save it in your project folders for future use:
+```bash
+curl https://raw.githubusercontent.com/paritytech/polkadot-sdk/refs/tags/polkadot-stable2412/substrate/.maintain/frame-weight-template.hbs \
+--output ./pallets/benchmarking/frame-weight-template.hbs
+```
+
+4. Run the benchmarking tool to measure extrinsic weights:
+
+    ```bash
+    frame-omni-bencher v1 benchmark pallet \
     --runtime INSERT_PATH_TO_WASM_RUNTIME \
     --pallet INSERT_NAME_OF_PALLET \
-    --extrinsic '*' \
-    --steps 20 \
-    --repeat 10 \
+    --extrinsic "" \
+    --template ./frame-weight-template.hbs \
     --output weights.rs
     ```
 
     !!! info "Flag definitions"
         - `--runtime` - the path to your runtime's Wasm
         - `--pallet` - the name of the pallet you wish to benchmark. This pallet must be configured in your runtime and defined in `define_benchmarks`
-        - `--extrinsic` - which extrinsic to test. Using `'*'` implies all extrinsics will be benchmarked
+        - `--extrinsic` - which extrinsic to test. Using `""` implies all extrinsics will be benchmarked
+        - `--template` - defines how weight information should be formatted
         - `--output` - where the output of the auto-generated weights will reside
 
-The generated `weights.rs` file contains weight annotations for your extrinsics, ready to be added to your pallet. The output should be similar to the following. Some output is omitted for brevity: 
+The generated `weights.rs` file contains weight annotations for your extrinsics, ready to be added to your pallet. The output should be similar to the following. Some output is omitted for brevity:
 
 --8<-- 'code/develop/parachains/testing/benchmarking/benchmark-output.html'
 
 #### Add Benchmark Weights to Pallet
 
-Once the `weights.rs` is generated, you may add the generated weights to your pallet. It is common that `weights.rs` become part of your pallet's root in `src/`:
+Once the `weights.rs` is generated, you need to integrate it with your pallet. This involves two steps: importing the `weights` module and the `WeightInfo` trait, and adding it to your pallet's `Config` trait. Here's how to set this up:
```

Comment:
```suggestion
Once the `weights.rs` is generated, you must integrate it with your pallet. To begin the integration, import the `weights` module and the `WeightInfo` trait, then add both to your pallet's `Config` trait. Complete the following steps to set up the configuration:
```
Please add a bullet here for "1. (describe this first step) above the first rust code snippet